drivers/volume/portworx/portworx.go

```diff
@@ -2608,7 +2615,59 @@ func (p *portworx) CancelMigration(migration *storkapi.Migration) error {
 func (p *portworx) UpdateMigratedPersistentVolumeSpec(
 	pv *v1.PersistentVolume,
 	vInfo *storkapi.ApplicationRestoreVolumeInfo,
+	namespaceMapping map[string]string,
 ) (*v1.PersistentVolume, error) {
+	// Get the pv storageclass and get the provision detail and decide on csi section.
+	if len(pv.Spec.StorageClassName) != 0 {
+		sc, err := storage.Instance().GetStorageClass(pv.Spec.StorageClassName)
+		if err != nil {
+			return nil, fmt.Errorf("failed in getting the storage class [%v]: %v", pv.Spec.StorageClassName, err)
+		}
+		if isCsiProvisioner(sc.Provisioner) {
+			// add csi section in the pv spec
+			if pv.Spec.CSI == nil {
+				pv.Spec.CSI = &v1.CSIPersistentVolumeSource{}
+			}
+			// get the destinationNamespace
+			var dstNamespace string
+			var exists bool
+			if dstNamespace, exists = namespaceMapping[vInfo.SourceNamespace]; !exists {
+				dstNamespace = vInfo.SourceNamespace
+			}
+			// Update the controller expand secret
+			if pv.Spec.CSI.ControllerExpandSecretRef == nil {
+				pv.Spec.CSI.ControllerExpandSecretRef = &v1.SecretReference{}
+			}
+			if val, ok := sc.Parameters[controllerExpandSecretName]; ok && val == templatizedName {
+				pv.Spec.CSI.ControllerExpandSecretRef.Name = vInfo.PersistentVolumeClaim
+			} else {
+				pv.Spec.CSI.ControllerExpandSecretRef.Name = val
+			}
+			// In the case of portworx volume backup, we will have namespace mapping always.
+			// So no need to check for the template string as we are going to change the value to destination namepsace always.
+			// If user does not change the namespace, the source and destionation namespacein the mapping will be same.
+			if _, ok := sc.Parameters[controllerExpandSecretNamespace]; ok {
+				pv.Spec.CSI.ControllerExpandSecretRef.Namespace = dstNamespace
+			}
+
+			// Update the node publish secret
+			if pv.Spec.CSI.NodePublishSecretRef == nil {
```

Comment:
same here: only set the NodePublishSecretRef object if you find those in the StorageClass parameters.

Reply:
This will ensure we don't set the "Namespace"  field in *SecretRef object even if there are not security params specified in the storage class.

Reply:
Fixed as suggested.